src/core/server/plugins/plugins_service.ts

```diff
@@ -135,13 +175,15 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
 
   public async start(deps: PluginsServiceStartDeps) {
     this.log.debug('Plugins service starts plugins');
-    const contracts = await this.pluginsSystem.startPlugins(deps);
+    const contracts = await this.pluginsSystem.startPlugins(PluginType.standard, deps);
+    await this.pluginsSystem.stopPlugins(PluginType.preboot);
     return { contracts };
   }
 
   public async stop() {
     this.log.debug('Stopping plugins service');
-    await this.pluginsSystem.stopPlugins();
+    await this.pluginsSystem.stopPlugins(PluginType.preboot);
+    await this.pluginsSystem.stopPlugins(PluginType.standard);
```

Comment:
Let me see how it'd look like. In the worst case, if you don't mind, I'll keep it as is since PluginSystem is supposed to handle multiple `stopPlugins` call correctly.

Reply:
> Is it possible due to pop() usage here 

Yes, I was referring to the fact that `stop` cleans up internal plugins list.

> It's not obvious and I don't see any tests verifying this behavior.

Okay, let me improve that then.

Reply:
Handled in https://github.com/elastic/kibana/pull/103636/commits/36f6eef7c67dad4de5bfc952310fa4146bf03864